tree/tree/inc/TBasket.h

```diff
@@ -108,10 +113,8 @@ class TBasket : public TKey {
    TBasket(const char *name, const char *title, TBranch *branch);
    virtual ~TBasket();
 
-           void    AdoptBuffer(TBuffer *user_buffer);
    virtual void    AdjustSize(Int_t newsize);
    virtual void    DeleteEntryOffset();
-           void    DisownBuffer();
```

Comment:
This is a backward-incompatible change in interface. I'm not sure it's a good idea. For example, why did you need to declare `TBranch` as a friend after the change in interface?

Reply:
Actually, this method was introduced in this branch - it doesn't exist currently.

The change was motivated by a comment from @pcanal - the `Adopt` / `Disown` methods are a specific protocol between `TBranch` and its child `TBasket`.  They would be difficult to use outside the `TBranch`.

To me, it made more sense for `TBranch` / `TBasket` to be friends rather than expose memory ownership concerns to anyone invoking ROOT code.

Would welcome feedback on the approach!